app/src/main/java/com/nervousfish/nervousfish/activities/WaitActivity.java

```diff
@@ -0,0 +1,151 @@
+ * to complete the pairing session.
+ */
+public final class WaitActivity extends Activity {
+    private static final Logger LOGGER = LoggerFactory.getLogger("WaitActivity");
+    private IServiceLocator serviceLocator;
+    private Object dataReceived;
+    private Object tapCombination;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        this.setContentView(R.layout.activity_wait);
+
+        final Intent intent = getIntent();
+        this.serviceLocator = (IServiceLocator) intent.getSerializableExtra(ConstantKeywords.SERVICE_LOCATOR);
+
+        this.serviceLocator.registerToEventBus(this);
+
+        this.dataReceived = intent.getSerializableExtra(ConstantKeywords.DATA_RECEIVED);
+        tapCombination = intent.getSerializableExtra(ConstantKeywords.TAP_DATA);
+
+        LOGGER.info("dataReceived is not null: " + (this.dataReceived != null)
+                + " tapCombination is not null: " + (tapCombination != null));
+
+        final String message = (String) intent.getSerializableExtra(ConstantKeywords.WAIT_MESSAGE);
+        final TextView waitingMessage = (TextView) findViewById(R.id.waiting_message);
+        waitingMessage.setText(message);
+
+        LOGGER.info("WaitActivity created");
+    }
+
+    private void evaluateData() {
+        //TODO: check if when we decrypt the dataReceived with the tapCombination that we get a normal contact
+        LOGGER.info("Evaluating data");
+        final Intent intent = new Intent(this, MainActivity.class);
+        intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, serviceLocator);
+        intent.putExtra(ConstantKeywords.SUCCESSFUL_BLUETOOTH, true);
+        this.startActivity(intent);
+        //setResult(ConstantKeywords.DONE_PAIRING_RESULT_CODE);
+        //finish();
+
+    }
+
+    /**
+     * Can be called by a button to cancel the pairing
+     * @param view The view that called this method
+     */
+    public void cancelWaiting(final View view) {
+        setResult(ConstantKeywords.CANCEL_PAIRING_RESULT_CODE);
+        finish();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void onActivityResult(final int requestCode, final int resultCode, final Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (resultCode == ConstantKeywords.DONE_PAIRING_RESULT_CODE) {
+            this.setResult(ConstantKeywords.DONE_PAIRING_RESULT_CODE);
+            finish();
+        } else if (resultCode == ConstantKeywords.CANCEL_PAIRING_RESULT_CODE) {
+            this.setResult(ConstantKeywords.CANCEL_PAIRING_RESULT_CODE);
+            finish();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void onStart() {
+        super.onStart();
+        if (this.dataReceived != null && tapCombination != null) {
+            evaluateData();
```

Comment:
Isn't `this.dataReceived != null && tapCombination != null` part of evaluating the data?